cmake/modules/RootMacros.cmake

```diff
@@ -266,7 +266,7 @@ endfunction(ROOT_GET_INSTALL_DIR)
 #---------------------------------------------------------------------------------------------------
 function(ROOT_GENERATE_DICTIONARY dictionary)
   CMAKE_PARSE_ARGUMENTS(ARG "STAGE1;MULTIDICT;NOINSTALL;NO_CXXMODULE"
-    "MODULE;LINKDEF" "NODEPHEADERS;OPTIONS;DEPENDENCIES;EXTRA_DEPENDENCIES;BUILTINS" ${ARGN})
+    "MODULE;LINKDEF" "NODEPHEADERS;OPTIONS;DEPENDENCIES;EXTRA_DEPENDENCIES;BUILTINS;INCLUDES" ${ARGN})
```

Comment:
Why do we need to specify the include list using an external argument `INCLUDES`. We seem to have all information necessary to deduce it?

Reply:
I add `INCLUDES` argument in several ROOT macros while now there are quite some include directories should be specified for each library. And during dictionary generation all these includes are required. One still can use old way - set `target_inlcude_directories`

Reply:
I would prefer to use the target_include_directories because this gives a more uniform view between compilation time (when the library is built) and runtime (when the dictionary is used). I do not want to give an opportunity for the users to put dictionary headers which are not part of the library which it link against.

Reply:
But `INCLUDES` argument is just full equivalent of `target_include_directories` cmake function. I introduce it to have possibility to specify include paths for library objects and dictionaty generator in consistent way.  User can tweak dictionary generation this or that way anyway.

Reply:
Another remark - `INCLUDES` argument in `ROOT_GENERATE_DICTIONARY` macro used only for ROOT dictionaries

Reply:
> But `INCLUDES` argument is just full equivalent of `target_include_directories` cmake function. I introduce it to have possibility to specify include paths for library objects and dictionaty generator in consistent way. User can tweak dictionary generation this or that way anyway.

If it is called from `ROOT_STANDARD_LIBRARY_PACKAGE` you can guarantee consistency. However, we can have separate calls to `ROOT_LINKER_LIBRARY` (or `add_library`) and then a call to `ROOT_GENERATE_DICTIONARY`. In this case the user will be able to specify different set of include paths. Although this can be considered a feature, in my opinion, it is a dangerous one and I'd like to really understand what would be the use case. Otherwise I am all for inheriting the `target_include_directories`.


> Another remark - `INCLUDES` argument in `ROOT_GENERATE_DICTIONARY` macro used only for ROOT dictionaries

Once we add this, we open it to the public which makes it very hard to remove.  See discussion in PR #5019 which is about `REFLEX_GENERATE_DICTIONARY` but expresses the same concerns.

Reply:
`INCLUDES` parameter simply ignored when used not in ROOT project. If you like, I can simply not parse it when used outside ROOT
Use of `target_include_directories` is still possible and supported.
But when one uses `target_include_directories` one should specify target second or third time - therefore I want to have `INCLUDES` argument in `ROOT_GENERATE_DICTIONARY` and `ROOT_STANDARD_LIBRARY_PACKAGE` directly. But only for ROOT libraries and dictionaries

Reply:
The dictionary knows which library it is built for, and can make use of the library's include directories. Why is this insufficient, why do we need an extra `INCLUDES` parameter?